clients/drcachesim/analyzer.cpp

```diff
@@ -622,6 +622,7 @@ analyzer_tmpl_t<RecordType, ReaderType>::process_tasks(analyzer_worker_data_t *w
                 worker->error =
                     "Failed to read from trace: " + worker->stream->get_stream_name();
             }
+            worker->stream->set_active(false); // Avoid hang in scheduler.
```

Comment:
This may be harder to accomplish so a comment would be sufficient for now.

Reply:
I don't think we know whether the same input on some other output would cause a problem.  This could be a shard-specific error.

For short-circuiting: first, there's no simple way other than a global exit/abort (or std:: versions).  That is a different model than the current one where error message are propagated to the caller -- so I'm not sure we want to go that way.  Maybe there are use cases where one shard's error is not fatal to others or the overall use case; even if not we'd have to make sure want analyzers just blowing up the whole process (hard to unit test errors, for one thing).

Reply:
Added a comment about this.

Reply:
> Maybe there are use cases where one shard's error is not fatal to others or the overall use case

You mean we may want to continue analyzing the trace and printing the results even though the tool says there was an error somewhere? At least right now we don't do that: after the threads join, the error will be thrown.

If some error is tolerable, perhaps the tool shouldn't return an error/false.

> I don't think we know whether the same input on some other output would cause a problem

I'm not sure whether we'd want to at all continue analyzing the trace past any error. But say we do: a more high level question you highlight is: when an error happens on the tool side at some memref, should the output be marked as bad, or the specific input stream that was active at that point? Here we are marking the output as bad but we cannot be sure.

Reply:
These other questions of an early global abort seem beyond the scope of this PR: this PR is only trying to avoid a hang, within the current error reporting structure.  If you would like to change the error reporting, maybe file an issue.  Currently the analyzer propagates errors to the caller.  The drcachesim launcher turns an analyzer error into a process exit, but other users might not (especially unit tests, which do not want that).